templates/_macros/hero.jinja

```diff
@@ -0,0 +1,127 @@
+# Params
+# header_text: Hero header text (required)
+# subheader_text: Hero subheader text
+# layout: layout of hero section. Options are '50/50', '75/25', '25/75', 'fallback'
+# layout_responsive_on_tablet: whether the layout should be applied to the tablet size specifically. Applies "-on-medium" to row class in 50/50 and 25/75 layouts
+# layout_responsive_on_desktop: whether the layout should be applied to the desktop size specifically. Applies "-on-large" to row class in 50/50 and 25/75 layouts
+# image_full_cover: whether the image should span full width of the hero. Only supported on 50/50 layout
+# cta_primary_url: primary CTA button URL
+# cta_secondary_url: secondary CTA button URL
+# Slots
+# description: paragraph-style content below the header and subheader
+# cta_primary: contents of the primary CTA button
+# cta_secondary: contents of the secondary CTA button
+# image: slot for image content (required)
+
+{% macro hero(
+  header_text="",
+  subheader_text="",
+  layout="fallback",
+  layout_responsive_on_tablet = false,
+  layout_responsive_on_desktop = true,
+  image_full_cover = false,
+  cta_primary_url = "#",
+  cta_secondary_url = "#"
+)-%}
+  {% set has_subheader = subheader_text|trim|length > 0 %}
+  {% set description_content = caller('description') %}
+  {% set has_description = description_content|trim|length > 0 %}
+  {% set cta_primary_content = caller('cta_primary') %}
+  {% set cta_secondary_content = caller('cta_secondary') %}
+  {% set has_cta = cta_primary_content|trim|length > 0 %}
+  {% set has_cta_secondary = cta_secondary_content|trim|length > 0 %}
+  {% set image_content = caller('image') %}
+
+  {# User can pass layout as "X-Y" or "X/Y" #}
+  {% set layout = layout | trim | replace('/', '-') %}
+  {% if layout not in ['50-50', '75-25', '25-75', 'fallback'] %}
+    {% set layout = 'fallback' %}
+  {% endif %}
+
+  {# Full cover image only supported on 50-50 layout #}
+  {% set image_full_cover = image_full_cover and layout == '50-50' %}
+
+  {# Construct grid classes from params input #}
+  {% set row_classes = [] %}
+  {% set col_classes = ['col', 'col'] %}
+  {% if layout in ['50-50', '25-75'] %}
+    {# Responsive breakpoint classes were requested #}
+    {% if layout_responsive_on_tablet or layout_responsive_on_desktop %}
+      {% if layout_responsive_on_tablet %}
+        {% set _ = row_classes.append('row--' + layout + '-on-medium') %}
```

Comment:
This is a way to append to a list without adding "None" to the rendered markup.

I've replaced it with `row_classes.append() or ""`, which gets rid of the placeholder LHS (but has to use an `or ""` to [write an empty string](https://stackoverflow.com/questions/71850614/jinja2-prints-none-when-appending-to-a-loop) rather than `None`